### ISSUE TITLE
[CI] SSLDriverTests failures on Java 11

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#32122]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java11,nodes=virtual&&linux/181/

These test failures reproduce on Java 11 only with a range of different errors
```
ERROR   0.66s | SSLDriverTests.testCloseDuringHandshake <<< FAILURES!
   > Throwable #1: javax.net.ssl.SSLException: Cannot kickstart, the connection is broken or closed
   >    at __randomizedtesting.SeedInfo.seed([2728A87A7764842:5CF8D7186D9C55B5]:0)
   >    at java.base/sun.security.ssl.TransportContext.kickstart(TransportContext.java:205)
   >    at java.base/sun.security.ssl.SSLEngineImpl.writeRecord(SSLEngineImpl.java:167)
   >    at java.base/sun.security.ssl.SSLEngineImpl.wrap(SSLEngineImpl.java:138)
   >    at java.base/sun.security.ssl.SSLEngineImpl.wrap(SSLEngineImpl.java:116)
   >    at java.base/javax.net.ssl.SSLEngine.wrap(SSLEngine.java:511)
```
```
FAILURE 0.50s | SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch <<< FAILURES!
   > Throwable #1: org.junit.ComparisonFailure: expected:<[Client requested protocol TLSv1 not enabled or not supported]> but was:<[Received fatal alert: protocol_version]>
   >    at __randomizedtesting.SeedInfo.seed([2728A87A7764842:9EE7D816738689EA]:0)
   >    at org.elasticsearch.xpack.security.transport.nio.SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch(SSLDriverTests.java:130)
```

Reproduce commands:
```
./gradlew :x-pack:plugin:security:test \
  -Dtests.seed=2728A87A7764842 \
  -Dtests.class=org.elasticsearch.xpack.security.transport.nio.SSLDriverTests \
  -Dtests.method="testHandshakeFailureBecauseNoCiphers" \
  -Dtests.security.manager=true \
  -Dtests.locale=bs-Latn-BA \
  -Dtests.timezone=Africa/Tunis

./gradlew :x-pack:plugin:security:test \
  -Dtests.seed=2728A87A7764842 \
  -Dtests.class=org.elasticsearch.xpack.security.transport.nio.SSLDriverTests \
  -Dtests.method="testCloseDuringHandshake" \
  -Dtests.security.manager=true \
  -Dtests.locale=bs-Latn-BA \
  -Dtests.timezone=Africa/Tunis

./gradlew :x-pack:plugin:security:test \
  -Dtests.seed=2728A87A7764842 \
  -Dtests.class=org.elasticsearch.xpack.security.transport.nio.SSLDriverTests \
  -Dtests.method="testRenegotiate" \
  -Dtests.security.manager=true \
  -Dtests.locale=bs-Latn-BA \
  -Dtests.timezone=Africa/Tunis

./gradlew :x-pack:plugin:security:test \
  -Dtests.seed=2728A87A7764842 \
  -Dtests.class=org.elasticsearch.xpack.security.transport.nio.SSLDriverTests \
  -Dtests.method="testHandshakeFailureBecauseProtocolMismatch" \
  -Dtests.security.manager=true \
  -Dtests.locale=bs-Latn-BA \
  -Dtests.timezone=Africa/Tunis

./gradlew :x-pack:plugin:security:test \
  -Dtests.seed=2728A87A7764842 \
  -Dtests.class=org.elasticsearch.xpack.security.transport.nio.SimpleSecurityNioTransportTests \
  -Dtests.method="testRenegotiation" \
  -Dtests.security.manager=true \
  -Dtests.locale=zh \
  -Dtests.timezone=America/Antigua

./gradlew :x-pack:plugin:security:test \
  -Dtests.seed=2728A87A7764842 \
  -Dtests.class=org.elasticsearch.xpack.security.transport.nio.SimpleSecurityNioTransportTests \
  -Dtests.security.manager=true \
  -Dtests.locale=en-US \
  -Dtests.timezone=Etc/UTC
```

**Comment by tvernum** [comment]
Ping: @tbrooks8

**Comment by tbrooks8** [comment]
Closed by #32145. #32144 is open for the one test that is muted.
